packages/common-integration-tests/src/auth-client.ts

```diff
@@ -336,6 +351,207 @@ export function runAuthClientTests(
+        )
+      ).toBeInstanceOf(TopicPublish.Success);
+      expect(
+        await sessionTokenTopicClient.publish(
+          FGA_CACHE_1,
+          FGA_TOPIC_2,
+          'This weird trick will make you a better developer...'
+        )
+      ).toBeInstanceOf(TopicPublish.Success);
+    });
+
+    it('can only read all caches', async () => {
+      const readAllCachesTokenResponse =
+        await sessionTokenAuthClient.generateAuthToken(
+          new Permissions([new CachePermission(CacheRole.ReadOnly)]),
+          ExpiresIn.seconds(10)
+        );
+      expect(readAllCachesTokenResponse).toBeInstanceOf(
+        GenerateAuthToken.Success
+      );
+      const readAllCachesToken = (
+        readAllCachesTokenResponse as GenerateAuthToken.Success
+      ).authToken;
+      const cacheClient = cacheClientFactory(readAllCachesToken);
+      // 1. Sets should fail
+      const setError1 = await cacheClient.set(FGA_CACHE_1, 'homer', 'simpson');
+      expect(setError1).toBeInstanceOf(CacheSet.Error);
+      expect(setError1).toHaveProperty(
+        'errorCode',
+        MomentoErrorCode.PERMISSION_ERROR
+      );
+      const setError2 = await cacheClient.set(FGA_CACHE_2, 'oye', 'caramba', {
+        ttl: 30,
+      });
+      expect(setError2).toBeInstanceOf(CacheSet.Error);
+      expect(setError2).toHaveProperty(
+        'errorCode',
+        MomentoErrorCode.PERMISSION_ERROR
+      );
+
+      // 2. Gets for existing keys should succeed with hits
+      const hitResp1 = await cacheClient.get(FGA_CACHE_1, FGA_CACHE_1_KEY);
+      expect(hitResp1).toBeInstanceOf(CacheGet.Hit);
+      expect(hitResp1).toHaveProperty('valueString', FGA_CACHE_1_VALUE);
+      const hitResp2 = await cacheClient.get(FGA_CACHE_2, FGA_CACHE_2_KEY);
+      expect(hitResp2).toBeInstanceOf(CacheGet.Hit);
+      expect(hitResp2).toHaveProperty('valueString', FGA_CACHE_2_VALUE);
+
+      // 3. Gets for non-existing keys return misses
+      const missResp1 = await cacheClient.get(FGA_CACHE_1, 'i-exist-not');
+      expect(missResp1).toBeInstanceOf(CacheGet.Miss);
+      const missResp2 = await cacheClient.get(FGA_CACHE_2, 'i-exist-not');
+      expect(missResp2).toBeInstanceOf(CacheGet.Miss);
+
+      const topicClient = topicClientFactory(readAllCachesToken);
+      const pubResp = await topicClient.publish(
+        FGA_CACHE_1,
+        FGA_TOPIC_1,
+        'breaking news!'
+      );
+      expect(pubResp).toBeInstanceOf(TopicPublish.Error);
+      expect(pubResp).toHaveProperty(
+        'errorCode',
+        MomentoErrorCode.PERMISSION_ERROR
+      );
+
+      const subResp = await topicClient.subscribe(
+        FGA_CACHE_1,
+        FGA_TOPIC_1,
+        trivialHandlers
+      );
+      expect(subResp).toBeInstanceOf(TopicSubscribe.Error);
+      expect(subResp).toHaveProperty(
+        'errorCode',
+        MomentoErrorCode.PERMISSION_ERROR
+      );
+    });
+
+    it('can read/write cache foo and all topics in cache bar', async () => {
```

Comment:
nit: reading these tests I'm having a little trouble mapping "foo" and "bar" to FGA_CACHE_* without scrolling back and forth

Reply:
Fixed in https://github.com/momentohq/client-sdk-javascript/pull/598/commits/f2aad35500cbec9f6c42ca858e70abafffacbf58